tests/ui/cast/ptr-to-trait-obj-wrap.rs

```diff
@@ -0,0 +1,29 @@
+//@ check-pass
+#![allow(unused)]
+
+trait A {}
+
+struct W<T: ?Sized>(T);
+struct X<T: ?Sized>(T);
+
+fn unwrap(a: *const W<dyn A>) -> *const dyn A {
```

Comment:
can you add tests that this kind of cast isnt allowed to add autotraits or change lifetime/generic parameters

Reply:
Added a test for that :)

Reply:
Added such tests.